Logic/logic_manager.cpp

```diff
@@ -8,46 +10,116 @@
 #include <thread>
 #include <unordered_map>
 #include <vector>
-// #include "src/nlohmann/json.hpp"
-// #include "concurrentqueue.h"
+
+#include "data_processing.hpp"
+#include "json_tools.hpp"
+#include "parser_connection_client.hpp"
+#include "storage_structures.hpp"
+#include "users_connections_server.hpp"
 
 using namespace boost::asio;
 
+moodycamel::ConcurrentQueue<std::string> p2p::parsers_responses;
+p2p::SharedString p2p::up_to_date_version;
+
+std::string p2p::PARSER_IP;
+uint16_t p2p::PARSER_PORT;
+uint16_t p2p::USERS_SERVER_PORT;
+bool p2p::PARSER_LOGS_ON;
+bool p2p::PARSER_RESPONSE_LOGS_ON;
+bool p2p::USERS_SERVER_LOGS_ON;
+bool p2p::ORDERS_FOR_BUY_LOGS_ON;
+bool p2p::ORDERS_FOR_SELL_LOGS_ON;
+bool p2p::SELL_BY_COIN_LOGS_ON;
+bool p2p::MARKET_RATES_LOGS_ON;
+bool p2p::CHAINS_LOGS_ON;
+
 namespace p2p {
 
-// Запускаем в отдельном потоке из Main::run
-void raise_parser_connection(std::string parser_connection_ip,
-                             std::string parser_connection_port) {
-  // Поднимаем клиента для подключения к парсеру:
-  boost::asio::io_context client_io_context;
-  ip::tcp::resolver resolver(client_io_context);
-  auto endpoints =
-      resolver.resolve(parser_connection_ip, parser_connection_port);
-  ParserConnectionClient client(client_io_context, endpoints);
-  client_io_context.run();
+void raise_parser_connection_and_users_server() {
+  boost::asio::io_context global_io_context;
+  ip::tcp::resolver resolver(global_io_context);
+  auto endpoints = resolver.resolve(PARSER_IP, std::to_string(PARSER_PORT));
+  ParserConnectionClient client(global_io_context, endpoints);
+  auto users_server = UsersConnectionsServer(global_io_context);
+  users_server.listen(USERS_SERVER_PORT);
+  global_io_context.run();
 }
 
-// Запускаем в отдельном потоке из Main::run
-void raise_users_server(uint16_t users_server_port) {
-  // Поднимаем сервер для подключений клиентов:
-  auto users_server = UsersConnectionsServer();
-  users_server.listen(users_server_port);
-  users_server.run();
+void to_analysis() {
+  std::string response;
+  std::string last_response;
+  while (true) {
+    bool has_new_responses = false;
+    while (parsers_responses.try_dequeue(response)) {
+      last_response = response;
+      has_new_responses = true;
+    }
+    if (has_new_responses) {
+      if (PARSER_RESPONSE_LOGS_ON) {
+        std::cout << "Last response: " << last_response << std::endl;
+      }
+
+      Orders orders;
+      MarketRates market_rates;
+      if (!last_response.empty() && last_response[0] == '{') {
+        unpack_json(last_response, orders, market_rates);
+      } else {
+        if (PARSER_RESPONSE_LOGS_ON) {
+          std::cout << "Received empty response.\n";
```

Comment:
could be better to use `spdlog` or similar for logging and they have options to control logging level/disabling logging